web/permissions/permissions.go

```diff
@@ -0,0 +1,158 @@
+package permissions
+
+import (
+	"errors"
+	"fmt"
+	"net/http"
+	"strings"
+
+	"github.com/cozy/cozy-stack/pkg/consts"
+	"github.com/cozy/cozy-stack/pkg/couchdb"
+	"github.com/cozy/cozy-stack/pkg/crypto"
+	"github.com/cozy/cozy-stack/pkg/instance"
+	"github.com/cozy/cozy-stack/pkg/permissions"
+	"github.com/cozy/cozy-stack/web/middlewares"
+	"github.com/labstack/echo"
+	jwt "gopkg.in/dgrijalva/jwt-go.v3"
+)
+
+var readable = true
+var none = false
+
+var blackList = map[string]bool{
+	consts.Sessions:  none,
+	consts.Files:     readable,
```

Comment:
I'm not sure what you mean by blacklisting here. A permission like `io.cozy.files:ALL:io.cozy.files.music-dir` must be acceptable. It allows to use all the methods/verbs on `/files` for the music directory. On `/data`, it is still read-only, but it is not the permissions that enforce that.